homeassistant/components/schluter/climate.py

```diff
@@ -0,0 +1,169 @@
+        return self._coordinator.last_update_success
+
+    @property
+    def should_poll(self):
+        """Return if platform should poll."""
+        return False
+
+    @property
+    def supported_features(self):
+        """Return the list of supported features."""
+        return self._support_flags
+
+    @property
+    def unique_id(self):
+        """Return unique ID for this device."""
+        return self._serial_number
+
+    @property
+    def name(self):
+        """Return the name of the thermostat."""
+        return self._coordinator.data[self._serial_number].name
+
+    @property
+    def temperature_unit(self):
+        """Return the unit of measurement."""
+        return self._unit
+
+    @property
+    def current_temperature(self):
+        """Return the current temperature."""
+        return self._coordinator.data[self._serial_number].temperature
+
+    @property
+    def hvac_mode(self):
+        """Return current mode. Only heat available for floor thermostat."""
+        return HVAC_MODE_HEAT
+
+    @property
+    def hvac_action(self):
+        """Return current operation. Can only be heating or idle."""
+        return (
+            CURRENT_HVAC_HEAT
+            if self._coordinator.data[self._serial_number].is_heating
+            else CURRENT_HVAC_IDLE
+        )
+
+    @property
+    def target_temperature(self):
+        """Return the temperature we try to reach."""
+        return self._coordinator.data[self._serial_number].set_point_temp
+
+    @property
+    def hvac_modes(self):
+        """List of available operation modes."""
+        return [HVAC_MODE_HEAT]
+
+    @property
+    def min_temp(self):
+        """Identify min_temp in Schluter API."""
+        return self._coordinator.data[self._serial_number].min_temp
+
+    @property
+    def max_temp(self):
+        """Identify max_temp in Schluter API."""
+        return self._coordinator.data[self._serial_number].max_temp
+
+    async def async_set_hvac_mode(self, hvac_mode):
+        """Mode is always heating, so do nothing."""
+
+    def set_temperature(self, **kwargs):
+        """Set new target temperature."""
+        target_temp = None
+        target_temp = kwargs.get(ATTR_TEMPERATURE)
+        serial_number = self._coordinator.data[self._serial_number].serial_number
+        _LOGGER.debug("Setting thermostat temperature: %s", target_temp)
+
+        try:
+            if target_temp is not None:
+                self._api.set_temperature(self._session_id, serial_number, target_temp)
+        except RequestException as ex:
+            _LOGGER.error("An error occurred while setting temperature: %s", ex)
+
+    async def async_added_to_hass(self):
+        """When entity is added to hass."""
+        self._coordinator.async_add_listener(self.async_write_ha_state)
+
+    async def async_will_remove_from_hass(self):
+        """When entity will be removed from hass."""
+        self._coordinator.async_remove_listener(self.async_write_ha_state)
```

Comment:
If we also implement the `async_update` entity method and await `coordinator.async_request_refresh` we allow the user to schedule manual updates via the common entity update service.